modules/storages/spec/common/storages/peripherals/storage_interaction/one_drive/file_info_query_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+require_module_spec_helper
+
+RSpec.describe Storages::Peripherals::StorageInteraction::OneDrive::FileInfoQuery, :webmock do
+  include JsonResponseHelper
+
+  let(:storage) do
+    create(:one_drive_storage,
+           :with_oauth_client,
+           drive_id: 'b!-RIj2DuyvEyV1T4NlOaMHk8XkS_I8MdFlUCq1BlcjgmhRfAj3-Z8RY2VpuvV_tpd')
+  end
+  let(:user) { create(:user) }
+  let(:token) { create(:oauth_client_token, user:, oauth_client: storage.oauth_client) }
+  let(:file_id) { '01BYE5RZ5MYLM2SMX75ZBIPQZIHT6OAYPB' }
+  let(:not_found_json) { not_found_response }
+  let(:forbidden_json) { forbidden_response }
+
+  before do
+    stub_request(:get, "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/items/#{file_id}")
+      .with(headers: { 'Authorization' => "Bearer #{token.access_token}" })
+      .to_return(status: 200, body: read_json('folder_drive_item'), headers: {})
```

Comment:
🟠 [As you pointed out](https://github.com/opf/openproject/pull/13689#issuecomment-1717432225), good time to discuss adopting `VCR` here. Set up is simpler as we don't have to connect to a local docker container and can hit the graph API directly 👍🏾

Reply:
Sounds good 👍🏾